src/engine/cache_engine.c

```diff
@@ -164,12 +165,20 @@ bool ocf_fallback_pt_is_on(ocf_cache_t cache)
 #define SEQ_CUTOFF_FULL_MARGIN \
 		(OCF_TO_EVICTION_MIN + OCF_PENDING_EVICTION_LIMIT)
 
-static inline bool ocf_seq_cutoff_is_on(ocf_cache_t cache)
+/**
+ * @brief Check if sequential cutoff should be activated or not
+ *
+ * @param cache Cache instance
+ * @param clines Cachelines number that could be inserted with current request
+ *
+ * @return true if seq cutoff should be activated, false otherwise
+ */
+static inline bool ocf_seq_cutoff_is_on(ocf_cache_t cache, uint64_t clines)
 {
 	if (!ocf_cache_is_device_attached(cache))
 		return false;
 
-	return (ocf_freelist_num_free(cache->freelist) <=
+	return ((ocf_freelist_num_free(cache->freelist) - clines) <=
```

Comment:
This may cause signed integer underflow - would be better to move "-clines" to the right side of "<="